year-2022/day-06/src/main.rs

```diff
@@ -0,0 +1,77 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    const TEST_INPUT_1: &str = "mjqjpqmgbljsphdztnvjfqwrcgsmlb";
+    const TEST_INPUT_2: &str = "bvwbjplbgvbhsrlpgdmjqwftvncz";
+    const TEST_INPUT_3: &str = "nppdvjthqldpwncqszvftbrmjlhg";
+    const TEST_INPUT_4: &str = "nznrnfrfntjfmvfwmzdfjlvtqnbhcprsg";
+    const TEST_INPUT_5: &str = "zcfzfwzzqfrljwzlrfnpqdbhtmscgvjw";
+
+    #[test]
+    fn test_solution() {
+        assert_eq!(find_start_marker(TEST_INPUT_1, 4), 7);
+        assert_eq!(find_start_marker(TEST_INPUT_1, 14), 19);
+
+        assert_eq!(find_start_marker(TEST_INPUT_2, 4), 5);
+        assert_eq!(find_start_marker(TEST_INPUT_2, 14), 23);
+
+        assert_eq!(find_start_marker(TEST_INPUT_3, 4), 6);
+        assert_eq!(find_start_marker(TEST_INPUT_3, 14), 23);
+
+        assert_eq!(find_start_marker(TEST_INPUT_4, 4), 10);
+        assert_eq!(find_start_marker(TEST_INPUT_4, 14), 29);
+
+        assert_eq!(find_start_marker(TEST_INPUT_5, 4), 11);
+        assert_eq!(find_start_marker(TEST_INPUT_5, 14), 26);
+
+    }
+}
+
+fn are_all_chars_different(chars: &[char], size: usize) -> bool {
+    let mut index = 0;
+
+    while index < (size - 1) {
+        let mut jndex = index + 1;
+
+        while jndex < size {
+            if chars[index] == chars[jndex] {
+                return false;
+            }
+
+            jndex += 1;
+        }
+
+        index += 1;
+    }
```

Comment:
In Rust puoi usare i range un pò come su python:
```rust
for index in 0..(size - 1) {
    for jndex in index + 1..size {
        if chars[index] == chars[jndex] {
            return false;
        }
    }
}
```

Reply:
Il range può anche essere inclusivo Es: `0..=(size - 1)`

Reply:
Ottima osservazione; effettivamente per le soluzioni di altri giorni credo di aver già il `range` come hai descritto...  
Non ho idea e non ricordo perché, invece, qui ho usato i `while`! 😅

Credo sia il refuso di una qualche logica che pensavo di implementare e/o avevo abbozzato e alla fine rimosso...

Comunque correggo... Grazie! 🙃